api/v1alpha1/gateway_types.go

```diff
@@ -177,6 +183,40 @@ type ListenerTLS struct {
 	Options map[string]string `json:"options"`
 }
 
+// RouteSelector provides a mechanism to associate routes with this Gateway.
+type RouteSelector struct {
+	// Routes specifies what routes to associate with this Gateway. If defined,
+	// NamespaceSelector and LabelSelector are ignored and only the specified
+	// routes are associated with this Gateway.
+	//
+	// If undefined, all routes or routes matching NamespaceSelector and/or
+	// LabelSelector are associated with this Gateway.
+	//
+	// +optional
+	Routes []core.ObjectReference `json:"routes"`
```

Comment:
Should `Routes` be broken out to specific routes, i.e. `HTTPRouteRefs`? See https://github.com/kubernetes-sigs/service-apis/issues/50 for details.